### PR TITLE
Add support for Quad9 secure dns server

### PR DESCRIPTION
**Pull request by mterron**
See www.quad9.net

**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

_1_

---
_Add support for Quad9 DNS provider_

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -764,6 +764,7 @@ setDNS() {
       Norton ""
       Comodo ""
       DNSWatch ""
+      Quad9 ""
       Custom "")
   # In a whiptail dialog, show the options
   DNSchoices=$(whiptail --separate-output --menu "Select Upstream DNS Provider. To use your own, select Custom." ${r} ${c} 6 \
```
Comment (Mcat12) [inline automated install/basic-install.sh]: You should increment the `6` value so all options are made sure to be shown at once.

**Comment by mcattle** [comment]
I'd suggest also adding Quad9's [secure IPv6 address](https://www.quad9.net/#/faq#is-there-ipv6-support-for-quad9) **2620:fe::fe**

It may also be worth adding a "Quad9 Unsecure" option, with the IPv4 and IPv6 addresses of 9.9.9.10 and 2620:fe::10 respectively, but that could be a new pull request.

**Comment by mterron** [comment]
@mcattle , I can't find a place to add IPv6 DNS servers anywhere. It seems to me that the place where I added Quad9 only supports IPv4 addresses as per the rest of the options and there is no other screen to set up IPv6 DNS servers.

**Comment by mcattle** [comment]
@mterron I'd have to dig around the code, but the web UI for pi-hole includes checkboxes for IPv6 options for Google, OpenDNS, and DNS.WATCH, so I'm guessing they're in there somewhere.  It could just be a web UI thing, though.

**Comment by mcattle** [comment]
@mcattle Upon further reflection, it's probably not worth adding Quad9's unsecure addresses as options, as it goes against the value proposition of the Quad9 service.  Those addresses mainly exist to help test false positives from their secure resolvers.  Anyone who wants to use them could always add them manually.

I have also not found IPv6 defaults in Pi-hole itself, but have submitted a PR for Pi-hole's AdminLTE web UI to add Quad9 as an option.

**Comment by Mcat12** [comment]
You would add the IPv6 version in the same area that the IPv4 upstreams are set. However, we have not added IPv6 upstreams in the installer yet because it does not differentiate between IPv4 and IPv6 installs in that function at the moment.

**Comment by Mausy5043** [comment]
No option to select this from the WebUI?

**Comment by mcattle** [comment]
@Mausy5043 The change to the WebUI is here: https://github.com/pi-hole/AdminLTE/pull/619

**Pull request merged**